libs/core/src/lib/multi-input/multi-input.component.ts

```diff
@@ -350,7 +350,9 @@ export class MultiInputComponent implements
         if (checked) {
             this.selected.push(value);
         } else {
-            this.selected.splice(this.selected.indexOf(value), 1);
+          this.selected.forEach(token => {
```

Comment:
What is this? You iterate over an array and use splice on it to make it empty?

Reply:
Yes, because if they are holding control, it can select them in any weird order. So basically if anything is selected and they click delete it should delete it

Reply:
"Selected" in the context of the multi-input code just means that the token has been added to the multi input, not that it is currently colored blue and marked for deletion.  Which is super confusing because the name of the class that is applied to the token when it is colored blue is 'fd-token--selected'.  So I guess you could consider there being two layers of selection from the multi-input's point of view.  I'm working on a fix now and i'll push directly to this branch